Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalHandlerListener.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// This component ensures that input events are forwarded to this component when focus or gaze is not required.
+    /// </summary>
+    public abstract class InputSystemGlobalHandlerListener : MonoBehaviour
```

Comment:
I'd rather not have this class and let users do like we do in DefaultPrimaryPointerSelector:
```
        public void Initialize()
        {
            IMixedRealityInputSystem inputSystem;
            if (MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem))
            {
                inputSystem.RegisterHandler<IMixedRealityPointerHandler>(this);
            }
        }
```

Reply:
I wanted an easy way to migrate users to new API, if they used the old one -- just change inheritance to the new one, and it works. 
Also it hides async logic (which might go away) -- but it might go away.